### PR TITLE
Add a snapcraft.yml for building Snaps

### PR DESCRIPTION
**Pull request by piperswe**
The included snapcraft.yml can be used to build a Snap (the application format used by Canonical's modern package manager) out of Prism. If the project wants in the future, Prism can publish these Snaps to the Snap Store so the Prism Launcher can be installed through the Ubuntu Software app on vanilla Ubuntu.

I haven't registered the Snap name, so it's currently free for anyone to reserve. I'd suggest that a Prism developer register the name prismlauncher at https://snapcraft.io/register-snap to ensure the name belongs to the project, even if there are no plans of setting up CI to publish snaps in the short term.

I have also modified JavaUtils.cpp to be able to autodetect the Java versions included in the Snap, and added "*.snap" to the .gitignore so the compiled Snap isn't accidentally committed to the repository.

Signed-off-by: Piper McCorkle <contact@piperswe.me>

### CODE REVIEW

File: launcher/java/JavaUtils.cpp [hunk 1]
```diff
@@ -439,19 +439,28 @@ QList<QString> JavaUtils::FindJavaPaths()
...
     };
+    // java installed in a snap is installed in the standard directory, but underneath $SNAP
+    auto snap = qEnvironmentVariable("SNAP", "");
+    auto scanJavaDirs = [&](const QString & dirPath)
+    {
+        scanJavaDir(dirPath);
+        if (snap != "") {
+            scanJavaDir(snap + dirPath);
+        }
+    };
```
Comment (Scrumplex) [inline launcher/java/JavaUtils.cpp]:
```suggestion
    // java installed in a snap is installed in the standard directory, but underneath $SNAP
    auto snap = qEnvironmentVariable("SNAP");
    auto scanJavaDirs = [&](const QString & dirPath)
    {
        scanJavaDir(dirPath);
        if (!snap.isNull()) {
            scanJavaDir(snap + dirPath);
        }
    };
```

**Review by Scrumplex — APPROVED** [review]
Looks good. I also reserved the name. Thanks for the PR

File: .gitignore [hunk 2]
```diff
@@ -47,3 +47,6 @@ result/
 # Flatpak
 .flatpak-builder
 flatbuild
+
+# Snap
+*.snap
```
Comment (flowln) [inline .gitignore]: add new line pls

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    desktop: usr/share/applications/org.prismlauncher.PrismLauncher.desktop
+    command: usr/bin/prismlauncher
+    extensions:
+      - kde-neon
+    plugs:
+      - home
+      - opengl
+      - network
+      - network-bind
+      - audio-playback
```
Comment (flowln) [inline snap/snapcraft.yaml]: here too!

**Review by flowln — APPROVED** [review]
LGTM, but please add the new lines so we don't have weird diffs in the future :p

**Comment by Scrumplex** [comment]
Thanks!

**Pull request merged**